src/praisonai-agents/praisonaiagents/runtime/doctor_registry.py

```
@@ -62,18 +67,110 @@ def collect_all_findings(self, config: Dict[str, Any]) -> List[Finding]:
-        
-        return result
+                warnings.warn(f"Error collecting findings for rule '{rule.rule_id}': {e}")
+                continue
+
+            if not findings:
+                continue
+
+            before = copy.deepcopy(result)
+            try:
+                fixed = rule.apply_fix(before)
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)